### PR TITLE
Browser+LibWeb+WebContent: Add ability to inspect local storage

### PR DESCRIPTION
**Pull request by vkoskiv**
Shown below:

image

### CODE REVIEW

File: Userland/Libraries/LibWeb/LocalStorageObject.h [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/String.h>
+#include <LibIPC/Forward.h>
+
+namespace Web::LocalStorage {
```
Comment (awesomekling) [inline Userland/Libraries/LibWeb/LocalStorageObject.h:12]: In LibWeb, namespaces are typically organized by specification. There is no "local storage" specification, as web storage is part of the main HTML specification. For that reason, this should be in the `Web::HTML` namespace (and also in the `LibWeb/HTML/` directory).

Also, this doesn't really seem to map to an existing concept in the spec. Would it make sense to pass it across the IPC boundary as a simple `OrderedHashMap<String, String>` instead? That's what `Web::HTML::Storage` uses for internal storage currently anyway.

**Comment by vkoskiv** [comment]
I added a separate commit for the IPC stuff as it seemed like its own thing :^)

File: Userland/Applications/Browser/BrowserWindow.cpp [hunk 2]
```diff
@@ -546,6 +546,10 @@ void BrowserWindow::create_new_tab(URL url, bool activate)
         return m_cookie_jar.get_all_cookies();
     };
 
+    new_tab.on_get_local_storage_objects = [this]() {
+        return active_tab().m_web_content_view->get_local_storage_objects();
```
Comment (linusg) [inline Userland/Applications/Browser/BrowserWindow.cpp:550]: I feel like "objects" is the wrong term for this - they're no objects, they're key/value pairs, or simply "entries" (https://infra.spec.whatwg.org/#map-entry):

> To reorder a [Storage](https://html.spec.whatwg.org/multipage/webstorage.html#storage-2) object storage, reorder storage's [map](https://html.spec.whatwg.org/multipage/webstorage.html#concept-storage-map)'s [entries](https://infra.spec.whatwg.org/#map-entry) in an [implementation-defined](https://infra.spec.whatwg.org/#implementation-defined) manner.

and

> Each key/value pair is called an entry.

File: Userland/Applications/Browser/LocalStorageModel.h [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    void set_items(OrderedHashMap<String, String> map);
+    void clear_items();
+    virtual int row_count(GUI::ModelIndex const&) const override { return m_local_storage_objects.size(); }
+    virtual int column_count(GUI::ModelIndex const& = GUI::ModelIndex()) const override { return Column::__Count; }
+    virtual String column_name(int column) const override;
+    virtual GUI::ModelIndex index(int row, int column = 0, GUI::ModelIndex const& = GUI::ModelIndex()) const override;
+    virtual GUI::Variant data(GUI::ModelIndex const& index, GUI::ModelRole role = GUI::ModelRole::Display) const override;
+
+private:
+    OrderedHashMap<String, String> m_local_storage_objects;
```
Comment (linusg) [inline Userland/Applications/Browser/LocalStorageModel.h:30]: Same here

File: Userland/Applications/Browser/StorageWidget.cpp [hunk 4]
```diff
@@ -44,4 +58,14 @@ void StorageWidget::clear_cookies()
...
+void StorageWidget::set_local_storage_objects(OrderedHashMap<String, String> objects)
+{
+
+    m_local_storage_model->set_items(objects);
+}
+
+void StorageWidget::clear_local_storage_objects()
+{
+    m_local_storage_model->clear_items();
+}
```
Comment (linusg) [inline Userland/Applications/Browser/StorageWidget.cpp:70]: Please separate the closing namespace curly bracket with a newline

**Review by linusg — COMMENTED** [review]
I think we can improve the naming, rest LGTM. Glad the yakbait worked :)

File: Userland/Applications/Browser/StorageWidget.cpp (hunk 4 above)
Comment (vkoskiv) [reply Userland/Applications/Browser/StorageWidget.cpp:70]: Surprised `clang-format` didn't catch that!

File: Userland/Applications/Browser/BrowserWindow.cpp (hunk 2 above)
Comment (vkoskiv) [reply Userland/Applications/Browser/BrowserWindow.cpp]: Thanks for looking that up! I didn't even think to look at the spec for this.

**Comment by vkoskiv** [comment]
@linusg Done and done! Thanks for the review!

File: Userland/Applications/Browser/StorageWidget.cpp [hunk 5]
```diff
@@ -44,4 +58,15 @@ void StorageWidget::clear_cookies()
     m_cookies_model->clear_items();
 }
 
+void StorageWidget::set_local_storage_entries(OrderedHashMap<String, String> entries)
+{
+
```
Comment (linusg) [inline Userland/Applications/Browser/StorageWidget.cpp:63]: No empty line needed here

**Review by linusg — COMMENTED** [review]
Another tiny nitpick!

File: Userland/Applications/Browser/StorageWidget.cpp (hunk 4 above)
Comment (linusg) [reply Userland/Applications/Browser/StorageWidget.cpp:70]: Yeah, it's a bit limited in that regard. Someone should probably file a bug! :^)

File: Userland/Applications/Browser/StorageWidget.cpp (hunk 5 above)
Comment (vkoskiv) [reply Userland/Applications/Browser/StorageWidget.cpp]: Small details matter!

**Pull request merged**